classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -0,0 +1,411 @@
+		}
+		return $this->_cache;
+	}
+
+	/**
+	 * Constructor
+	 */
+	function DOIPubIdExportPlugin() {
+		parent::ImportExportPlugin();
+	}
+
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	function register($category, $path) {
+		$success = parent::register($category, $path);
+		if (!Config::getVar('general', 'installed')) return false;
+		if ($success) {
+			HookRegistry::register('AcronPlugin::parseCronTab', array($this, 'callbackParseCronTab'));
+		}
+		$this->addLocaleData();
+		return $success;
+	}
+
+	/**
+	 * @copydoc Plugin::getTemplatePath($inCore)
+	 */
+	function getTemplatePath($inCore = false) {
+		return parent::getTemplatePath($inCore) . 'templates/';
+	}
+
+	/**
+	 * @copydoc Plugin::manage()
+	 */
+	function manage($args, $request) {
+		$user = $request->getUser();
+		$router = $request->getRouter();
+		$context = $router->getContext($request);
+
+		$form = $this->_instantiateSettingsForm($context);
+		$notificationManager = new NotificationManager();
+		switch ($request->getUserVar('verb')) {
+			case 'save':
+				$form->readInputData();
+				if ($form->validate()) {
+					$form->execute();
+					$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS);
+					return new JSONMessage();
+				} else {
+					return new JSONMessage(true, $form->fetch($request));
+				}
+			default:
+				$form->initData();
+				return new JSONMessage(true, $form->fetch($request));
+		}
+		return parent::manage($args, $request);
+	}
+
+	/**
+	 * Get status names for the filter search option.
+	 * @param $detailedView boolean Detailed status view e.g. with a link to an agency page,
+	 * that contains more information about the DOI status
+	 * @return array (string status => string text)
+	 */
+	function getStatusNames() {
+		return array(
+			DOI_EXPORT_STATUS_ANY => __('plugins.importexport.common.status.any'),
+			DOI_EXPORT_STATUS_NOT_DEPOSITED => __('plugins.importexport.common.status.notDeposited'),
+			DOI_EXPORT_STATUS_MARKEDREGISTERED => __('plugins.importexport.common.status.markedRegistered'),
+		);
+	}
+
+	/**
+	 * Get status actions for the display to the user,
+	 * i.e. links to an agency with more information about the DOI status.
+	 * @param $pubObject
+	 * @return array (string status => link)
+	 */
+	function getStatusActions($pubObject) {
+		return array();
+	}
+
+	/**
+	 * Get action names for the action drop-down.
+	 * @return array (string action => string text)
+	 */
+	function getActionNames() {
+		return array(
+			DOI_EXPORT_ACTION_EXPORT => __('plugins.importexport.common.export'),
+			DOI_EXPORT_ACTION_MARKREGISTERED => __('plugins.importexport.common.markRegistered'),
+			DOI_EXPORT_ACTION_DEPOSIT => __('plugins.importexport.common.register'),
+			DOI_EXPORT_ACTION_CHECKSTATUS => __('plugins.importexport.common.checkStatus'),
+		);
+	}
+
+	/**
+	 * Get the plugin ID used as plugin settings prefix.
+	 * @return string
+	 */
+	abstract function getPluginId();
```

Comment:
`Id` suggests a numeric ID to me; maybe better to call it something else? e.g. `getPluginPrefix` or something?